crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -143,93 +297,201 @@ impl<F, T>
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: &DatatransRouterData<&types::RefundsRouterData<F>>,
     ) -> Result<Self, Self::Error> {
         Ok(Self {
             amount: item.amount.to_owned(),
+            currency: item.router_data.request.currency.to_string(),
+            refno: item.router_data.connector_request_reference_id.clone(),
         })
     }
 }
 
-// Type definition for Refund Response
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, DataTransRefundsResponse>>
+    for types::RefundsRouterData<api::Execute>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Execute, DataTransRefundsResponse>,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            DataTransRefundsResponse::Error(error) => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: error.code.unwrap_or(consts::NO_ERROR_CODE.to_string()),
+                    message: error
+                        .message
+                        .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                    reason: None,
+                    attempt_status: None,
+                    connector_transaction_id: None,
+                    status_code: item.http_code,
+                }),
+                ..item.data
+            }),
+            DataTransRefundsResponse::Success(response) => Ok(Self {
+                response: Ok(types::RefundsResponseData {
+                    connector_refund_id: response.transaction_id,
+                    refund_status: enums::RefundStatus::Success,
+                }),
+                ..item.data
+            }),
+        }
+    }
+}
 
-#[allow(dead_code)]
-#[derive(Debug, Serialize, Default, Deserialize, Clone)]
-pub enum RefundStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
+impl TryFrom<types::RefundsResponseRouterData<api::RSync, DataTransSyncResponse>>
+    for types::RefundsRouterData<api::RSync>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::RSync, DataTransSyncResponse>,
+    ) -> Result<Self, Self::Error> {
+        let response = match item.response {
+            DataTransSyncResponse::Error(error) => Err(types::ErrorResponse {
+                code: error.code.unwrap_or(consts::NO_ERROR_CODE.to_string()),
+                message: error
+                    .message
+                    .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                reason: None,
+                attempt_status: None,
+                connector_transaction_id: None,
+                status_code: item.http_code,
+            }),
+            DataTransSyncResponse::Response(response) => Ok(types::RefundsResponseData {
+                connector_refund_id: response.transaction_id.to_string(),
```

Comment:
Verify by printing once if you are receiving the correct `transaction_id` in RSYNC response

Reply:
That's the unique id generated every single time a refund is initiated . Its not the original transaction_id of the Payment